src/sentry/runner/commands/backup.py

```diff
@@ -16,9 +16,22 @@
 def import_(src):
     "Imports data from a Sentry export."
 
-    for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
-        if obj.object._meta.app_label not in EXCLUDED_APPS:
-            obj.save()
+    try:
+        with transaction.atomic():
+            for obj in serializers.deserialize("json", src, stream=True, use_natural_keys=True):
+                if obj.object._meta.app_label not in EXCLUDED_APPS:
+                    obj.save()
+    # For all database integrity errors, let's warn users to follow our
+    # recommended backup/restore workflow before reraising exception. Most of
+    # these errors come from restoring on a different version of Sentry or not restoring
+    # on a clean install.
+    except IntegrityError as e:
+        warningText = ">> Are you restoring from a backup of the same version of Sentry?\n>> Are you restoring onto a clean database?\n>> https://github.com/getsentry/sentry/issues/new/choose"
```

Comment:
Sorry I was unclear, if we have the link we should _also_ have the "if neither, then open an issue here:" text. 🐭

Reply:
ah added 😀

Reply:
🙈 https://github.com/getsentry/sentry/pull/44705/files#r1116394011